internal/server/storage/drivers/driver_linstor_volumes.go

```diff
@@ -0,0 +1,398 @@
+		return fmt.Errorf("Unable to get the Linstor client: %w", err)
+	}
+
+	if vol.contentType == ContentTypeFS {
+		// Create mountpoint.
+		err := vol.EnsureMountPath()
+		if err != nil {
+			return err
+		}
+
+		rev.Add(func() { _ = os.Remove(vol.MountPath()) })
+	}
+
+	// Transform byte to KiB.
+	requiredKiB, err := units.ParseByteSizeString(vol.ConfigSize())
+	if err != nil {
+		return fmt.Errorf("Unable to parse volume size: %w", err)
+	}
+
+	requiredKiB = requiredKiB / 1024
+	resourceDefinitionName := d.getResourceDefinitionName(vol)
+
+	// Create volume
+	err = linstor.Client.ResourceGroups.Spawn(context.TODO(), d.config[LinstorResourceGroupNameConfigKey], linstorClient.ResourceGroupSpawn{
+		ResourceDefinitionName: resourceDefinitionName,
+		VolumeSizes:            []int64{requiredKiB},
+	})
+	if err != nil {
+		return fmt.Errorf("Unable to spawn from resource group: %w", err)
+	}
+
+	l.Debug("Spawned a new Linstor resource definition for volume", logger.Ctx{"resourceDefinitionName": resourceDefinitionName})
+	rev.Add(func() { _ = d.DeleteVolume(vol, op) })
+
+	// Setup the filesystem
+	err = d.makeVolumeAvailable(vol)
+	if err != nil {
+		return fmt.Errorf("Could not make volume available for filesystem creation: %w", err)
+	}
+
+	devPath, err := d.getLinstorDevPath(vol)
+	if err != nil {
+		return fmt.Errorf("Could get device path for filesystem creation: %w", err)
+	}
+
+	volFilesystem := vol.ConfigBlockFilesystem()
+	if vol.contentType == ContentTypeFS {
+		_, err = makeFSType(devPath, volFilesystem, nil)
+		if err != nil {
+			return err
+		}
+	}
+
+	// For VMs, also create the filesystem volume.
+	if vol.IsVMBlock() {
+		fsVol := vol.NewVMBlockFilesystemVolume()
+
+		err := d.CreateVolume(fsVol, nil, op)
+		if err != nil {
+			return err
+		}
+
+		rev.Add(func() { _ = d.DeleteVolume(fsVol, op) })
+	}
+
+	err = vol.MountTask(func(mountPath string, op *operations.Operation) error {
+		// Run the volume filler function if supplied.
+		if filler != nil && filler.Fill != nil {
+			var err error
+			var devPath string
+
+			if IsContentBlock(vol.contentType) {
+				devPath, err = d.GetVolumeDiskPath(vol)
+				if err != nil {
+					return err
+				}
+			}
+
+			allowUnsafeResize := false
+
+			// Run the filler.
+			err = d.runFiller(vol, devPath, filler, allowUnsafeResize)
+			if err != nil {
+				return err
+			}
+
+			// Move the GPT alt header to end of disk if needed.
+			if vol.IsVMBlock() {
+				err = d.moveGPTAltHeader(devPath)
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+		return nil
+	}, op)
+	if err != nil {
+		return nil
+	}
+
+	rev.Success()
+	return nil
+}
+
+// DeleteVolume deletes a volume of the storage device.
+func (d *linstor) DeleteVolume(vol Volume, op *operations.Operation) error {
+	l := d.logger.AddContext(logger.Ctx{"volume": vol.Name()})
+	l.Debug("Deleting Linstor volume")
+
+	linstor, err := d.state.Linstor()
+	if err != nil {
+		return fmt.Errorf("Unable to get the Linstor client: %w", err)
```

Comment:
This adds a prefix that does not really add any information to an already prefixed error